src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -346,7 +346,13 @@ void GcsResourceManager::DeleteResources(
 void GcsResourceManager::OnNodeAdd(const rpc::GcsNodeInfo &node) {
   auto node_id = NodeID::FromBinary(node.node_id());
   if (!cluster_scheduling_resources_.contains(node_id)) {
-    cluster_scheduling_resources_.emplace(node_id, SchedulingResources());
+    std::unordered_map<std::string, double> resource_mapping;
+    for (auto &entry : node.resources_total()) {
+      resource_mapping.emplace(entry.first, entry.second);
+    }
```

Comment:
I'm not sure, but will this work?

`std::unordered_map<std::string, double> resource_mapping(node.resources_total().begin(), node.resources_total().end())`